pandas/core/indexes/multi.py

```diff
@@ -2156,6 +2156,8 @@ def _drop_from_level(self, codes, level, errors="raise"):
         i = self._get_level_number(level)
         index = self.levels[i]
         values = index.get_indexer(codes)
+        nan_codes = isna(codes)
```

Comment:
can't you just check `values==-1` ? as these are by definition nan.

Reply:
Unfortunately not, values is ``-1`` when the value to drop was not found in the index **and** when the value to drop is ``NaN`` and was found in the index. So we have to check, if it is not ``NaN`` before performing the isin.